### ISSUE TITLE
Info

### ISSUE THREAD

**Issue by Pigrecos** [MahdiSafsafi/UnivDisasm#9]
hello, sorry for my bad English.
I wanted to ask if you had planned to add support for specifying the type of the Argument access and support for specifying the read-write Implicit registers ,Very useful in the analysis of the detailed code.

These features are present in the capstone-next library.
In my local copy in your library I have added support to the type of access only for general purpose instructions

**Comment by MahdiSafsafi** [comment]
Hi Pigrecos,
Yep this feature is on my TODO-LIST, however it will be included only in a major version ...that means in version 2 since it needs to make a lot of change(building the instructions decoder from scratch).
When I started working on UnivDisasm I didn't find such info (at least in a parsable format). Now I have this info and all what I need is to build a stronger map first before writing the code.
Please, if you have any additional request/feature to ask ... feel free to post. This would be very helpful specially before releasing v2.
By the way, don't worry about your English ! I'm not a native English speaker too.

Thanks,
Mahdi.

**Comment by Pigrecos** [comment]
Thanks  ;)

**Comment by MahdiSafsafi** [comment]
@Pigrecos I see you did a good job with GP instructions (I mean including access mode feature). Just want to know where did you get access mode informations ?

**Comment by Pigrecos** [comment]
Hello, I got the information from the capstone Header, but they are also in the Intel manual.

Just a moment I'll tell you the precise references, why not remember the names :)

**Comment by MahdiSafsafi** [comment]
Thanks for your reply,
I know about Intel's docs and I've all the informations (including access mode) extracted from this docs ([Parsable-Instructions](https://github.com/MahdiSafsafi/Parsable-Instructions)). I just wanted to know if there's an alternative resource beyond Intel's docs.

**Comment by Pigrecos** [comment]
Hello, 
The capstone(next version) header file is 'X86MappingInsnOp.inc' 

:)

**Comment by Pigrecos** [comment]
https://github.com/asmjit/asmdb

This is a good resource, the types of access to the operands are also present.

Hello ;)

**Comment by MahdiSafsafi** [comment]
Ohh thank you man ! That's very helpful.
I was going to use https://github.com/MahdiSafsafi/Parsable-Instructions to rearrange all instructions.
How I didn't noticed such a great project !

**Comment by IL2** [comment]
Hi Mahdi! Please have a look at these docs https://github.com/intelxed/xed/tree/master/datafiles which come officially from Intel just recently.
via https://twitter.com/_rsc/status/810125713894469632

**Comment by MahdiSafsafi** [comment]
@IL2 Didn't know that ... Thank you !
Do you know anything about a similar thing to ARM ?

**Comment by IL2** [comment]
Glad that information came home, but I know nothing about ARM.
I can only guess, that it might be available in future just as Intel has done.
Is it a hint that you are to start working on ARM arch disasm?

**Comment by MahdiSafsafi** [comment]
@IL2 Yes, I already started working on ARM :) . I'm making a hard effort to parse ARM data from its pdf ... and that's something not easy compared to x86. I really wish if ARM people release a parsable format of instructions.

**Comment by IL2** [comment]
Aye @MahdiSafsafi, I see a little better importance of *machine-readable (non-PDF) description* of the arch commands might mean now and from the comments along that tweet.
Rejoice! :) One of the comments on Dec 17 is https://twitter.com/aramh/status/810126066903814144 is telling that ARM docs release is approved, and sooner or later will be available.

**Comment by MahdiSafsafi** [comment]
That'd be very very helpful !

**Comment by MahdiSafsafi** [comment]
@IL2 Take a look on this new project https://github.com/MahdiSafsafi/opcodesDB
Right now it only support AArch64!

**Comment by IL2** [comment]
@MahdiSafsafi Good news! Please have a look here https://twitter.com/maver/status/855243564586065921
https://developer.arm.com/products/architecture/a-profile/exploration-tools

**Comment by MahdiSafsafi** [comment]
@IL2, Thanks man ! I saw it two days ago :)
